kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1352 @@
+    #endif
+        }
+
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", ( this->GetEchoLevel() > 2)) << "Initializing element loop" << std::endl;
+
+        #pragma omp parallel firstprivate(dof_list)
+        {
+            // Gets the array of elements from the modeler
+            ElementsArrayType& r_elements_array = rModelPart.Elements();
+            const int number_of_elements = static_cast<int>(r_elements_array.size());
+            #pragma omp for schedule(guided, 512) nowait
+            for (int i = 0; i < number_of_elements; ++i) {
+                auto it_elem = r_elements_array.begin() + i;
+                const unsigned int this_thread_id = OpenMPUtils::ThisThread();
+
+                // Gets list of Dof involved on every element
+                pScheme->GetElementalDofList(*(it_elem.base()), dof_list, r_current_process_info);
+                dofs_aux_list_all[this_thread_id].insert(dof_list.begin(), dof_list.end());
+            }
+        }
+
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", ( this->GetEchoLevel() > 2)) << "Initializing condition loop" << std::endl;
+
+        #pragma omp parallel firstprivate(dof_list)
+        {
+            // Gets the array of conditions from the modeler
+            ConditionsArrayType& r_conditions_array = rModelPart.Conditions();
+            const int number_of_conditions = static_cast<int>(r_conditions_array.size());
+            #pragma omp for  schedule(guided, 512)
+            for (int i = 0; i < number_of_conditions; ++i) {
+                auto it_cond = r_conditions_array.begin() + i;
+                const unsigned int this_thread_id = OpenMPUtils::ThisThread();
+
+                // Gets list of Dof involved on every element
+                pScheme->GetConditionDofList(*(it_cond.base()), dof_list, r_current_process_info);
+                dofs_aux_list_all[this_thread_id].insert(dof_list.begin(), dof_list.end());
+            }
+        }
+
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", ( this->GetEchoLevel() > 2)) << "Initializing constraints loop" << std::endl;
+
+        #pragma omp parallel firstprivate(dof_list, auxiliar_dof_list)
+        {
+            auto& r_constraints_array = rModelPart.MasterSlaveConstraints();
+            const int number_of_constraints = static_cast<int>(r_constraints_array.size());
+            #pragma omp for  schedule(guided, 512)
+            for (int i = 0; i < number_of_constraints; ++i) {
+                auto it_const = r_constraints_array.begin() + i;
+                const IndexType this_thread_id = OpenMPUtils::ThisThread();
+
+                // Gets list of Dof involved on every element
+                it_const->GetDofList(dof_list, auxiliar_dof_list, r_current_process_info);
+                dofs_aux_list_all[this_thread_id].insert(dof_list.begin(), dof_list.end());
```

Comment:
this operation can be very very slow if you insert out of order...

Reply:
It is exactly the same as done in the other B&S